tests/test_utils.py

```diff
@@ -743,10 +746,322 @@ def test_projected_to_projected_transformer(self):
+        shape_path = os.path.join(self.workspace_dir, 'point_shape.shp')
+        # Create point shapefile to use as testing input
+        pygeoprocessing.shapely_geometry_to_vector(
+            geometries, shape_path, projection_wkt,
+            'ESRI Shapefile', fields=fields, attribute_list=attrs,
+            ogr_geom_type=ogr.wkbPoint)
+
+        shape_copy_path = os.path.join(
+            self.workspace_dir, 'point_shape_copy.shp')
+        # Create point shapefile to use as testing input
+        pygeoprocessing.shapely_geometry_to_vector(
+            geometries_copy, shape_copy_path, projection_wkt,
+            'ESRI Shapefile', fields=fields, attribute_list=attrs_copy,
+            ogr_geom_type=ogr.wkbPoint)
+
+        result = utils._assert_vectors_equal(shape_path, shape_copy_path)
+        self.assertFalse(result[0])
+        self.assertTrue("Vector feature counts are not the same" in result[1])
+
+    def test_different_projections(self):
+        """Utils: test vectors w/ different projections fails."""
+        from natcap.invest import utils
+
+        # Setup parameters to create point shapefile
+        fields = {'id': ogr.OFTReal, 'foo': ogr.OFTReal}
+        attrs = [{'id': 1, 'foo': 2.3456}, {'id': 2, 'foo': 5.6789}]
+
+        srs = osr.SpatialReference()
+        srs.ImportFromEPSG(3157)
+        projection_wkt = srs.ExportToWkt()
+        origin = (443723.127327877911739, 4956546.905980412848294)
+        pos_x = origin[0]
+        pos_y = origin[1]
+
+        geometries = [Point(pos_x + 50, pos_y - 50),
+                      Point(pos_x + 50, pos_y - 150)]
+
+        srs_copy = osr.SpatialReference()
+        srs_copy.ImportFromEPSG(26910)  # UTM Zone 10N
+        projection_wkt_copy = srs_copy.ExportToWkt()
+
+        origin_copy = (1180000, 690000)
+        pos_x_copy = origin_copy[0]
+        pos_y_copy = origin_copy[1]
+
+        geometries_copy = [Point(pos_x_copy + 50, pos_y_copy - 50),
+                           Point(pos_x_copy + 50, pos_y_copy - 150)]
+
+        shape_path = os.path.join(self.workspace_dir, 'point_shape.shp')
+        # Create point shapefile to use as testing input
+        pygeoprocessing.shapely_geometry_to_vector(
+            geometries, shape_path, projection_wkt,
+            'ESRI Shapefile', fields=fields, attribute_list=attrs,
+            ogr_geom_type=ogr.wkbPoint)
+
+        shape_copy_path = os.path.join(
+            self.workspace_dir, 'point_shape_copy.shp')
+        # Create point shapefile to use as testing input
+        pygeoprocessing.shapely_geometry_to_vector(
+            geometries_copy, shape_copy_path, projection_wkt_copy,
+            'ESRI Shapefile', fields=fields, attribute_list=attrs,
+            ogr_geom_type=ogr.wkbPoint)
+
+        result = utils._assert_vectors_equal(shape_path, shape_copy_path)
+        self.assertFalse(result[0])
+        self.assertTrue("Vector projections are not the same" in result[1])

```

Comment:
Great tests! I noticed one thing that's not tested is catching geometries that are not equal. What do you think about adding one more? Or even re-purposing one of the existing geometry tests if you want to. I don't think it's super-critical to test different geometry types.

Reply:
Oops! Definitely should check that.

Reply:
Added a test where geometry is expected to fail here: https://github.com/natcap/invest/pull/150/commits/9777b24707ce158d386e4c0403c56eb15ddd2061